numba_cuda/numba/cuda/cuda_paths.py

```diff
@@ -47,20 +69,143 @@ def _get_nvvm_path_decision():
     options = [
         ("Conda environment", get_conda_ctk()),
         ("Conda environment (NVIDIA package)", get_nvidia_nvvm_ctk()),
+        ("NVIDIA NVCC Wheel", _get_nvvm_wheel()),
         ("CUDA_HOME", get_cuda_home(*_nvvm_lib_dir())),
         ("System", get_system_ctk(*_nvvm_lib_dir())),
     ]
+
     by, path = _find_valid_path(options)
     return by, path
 
 
+def _get_nvrtc_system_ctk():
+    sys_path = get_system_ctk("bin" if IS_WIN32 else "lib64")
+    candidates = find_lib("nvrtc", sys_path)
+    if candidates:
+        return max(candidates)
+
+
+def _get_nvrtc_path_decision():
+    options = _build_options(
+        [
+            ("CUDA_HOME", get_cuda_home("nvrtc")),
+            ("Conda environment", get_conda_ctk()),
+            ("NVIDIA NVCC Wheel", _get_nvrtc_wheel()),
+            ("System", _get_nvrtc_system_ctk()),
+        ]
+    )
+    return _find_valid_path(options)
+
+
+def _get_nvvm_wheel():
+    platform_map = {
+        "linux": ("lib64", "libnvvm.so"),
+        "win32": ("bin", "nvvm64_40_0.dll"),
+    }
+
+    for plat, (dso_dir, dso_path) in platform_map.items():
+        if sys.platform.startswith(plat):
+            break
+    else:
+        raise NotImplementedError("Unsupported platform")
+
+    site_paths = [site.getusersitepackages()] + site.getsitepackages()
+
+    for sp in filter(None, site_paths):
+        nvvm_path = Path(sp, "nvidia", "cuda_nvcc", "nvvm", dso_dir, dso_path)
+        if nvvm_path.exists():
+            return str(nvvm_path.parent)
+
+    return None
+
+
+def detect_nvrtc_major_cuda_version(lib_dir):
+    # TODO - is this a bad idea?
```

Comment:
I don't think detecting the cuda major version from a file suffix that presumably maps to the SOVERSION of that file suffix is a particularly good idea. CMake does a relatively good job of handling this and its process is documented here (though it's very nvcc centric): https://cmake.org/cmake/help/latest/module/FindCUDAToolkit.html